### PR TITLE
CIV-8291 Manual Determination

### PR DESCRIPTION
**Pull request by portizdev**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/CIV-8291

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by ElifDenizEsea — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java [hunk 1]
```diff
@@ -45,14 +49,39 @@ public List<CaseEvent> handledEvents() {
...
+        updateClaimEndState(response, updatedData);
+
+        return response.build();
+    }
+
+    private void updateClaimEndState(AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response, CaseData updatedData) {
+        if (updatedData.hasClaimantAgreedToFreeMediation()) {
+            response.state(CaseState.IN_MEDIATION.name());
+        } else if (updatedData.hasApplicantRejectedRepaymentPlan() && (updatedData.getRespondent1().isCompany() || updatedData.getRespondent1().isOrganisation())) {
+            response.state(CaseState.All_FINAL_ORDERS_ISSUED.name());
```
Comment (neeta-hmcts) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java:81]: For Manual Determination claim end state suppose to be 'PROCEEDS_IN_HERITAGE_SYSTEM' ??

Comment (portizdev) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java:81]: Finally yes, business decisions.

**Review by neeta-hmcts — DISMISSED** [review]

**Review by kenneth-hmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java [hunk 2]
```diff
@@ -62,11 +75,13 @@ private CallbackResponse aboutToSubmit(CallbackParams callbackParams) {
     }
 
     private void updateClaimEndState(AboutToStartOrSubmitCallbackResponse.AboutToStartOrSubmitCallbackResponseBuilder response, CaseData updatedData) {
-
         if (updatedData.hasClaimantAgreedToFreeMediation()) {
             response.state(CaseState.IN_MEDIATION.name());
+        } else if (updatedData.hasApplicantRejectedRepaymentPlan() && (updatedData.getRespondent1().isCompany() || updatedData.getRespondent1().isOrganisation())) {
```
Comment (jarekPierchala) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java:80]: any chance to put the condition to separate function?

**Review by kalachandrasekar1 — DISMISSED** [review]

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java (hunk 2 above)
Comment (portizdev) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/ClaimantResponseCuiCallbackHandler.java]: I have grouped the second part to make it shorter and more understandable.

**Review by ElifDenizEsea — APPROVED** [review]

**Review by laarmada — APPROVED** [review]

**Review by rajakm — APPROVED** [review]

**Pull request merged**